dm/syncer/checkpoint.go

```diff
@@ -616,12 +617,11 @@ func (cp *RemoteCheckPoint) IsOlderThanTablePoint(table *filter.Table, location
 		return false
 	}
 	oldLocation := point.MySQLLocation()
-	cp.logCtx.L().Debug("compare table location whether is newer", zap.Stringer("location", location), zap.Stringer("old location", oldLocation))
+	// if we update enable-gtid = false to true, we need to compare binlog position instead of GTID before we save table point
+	cmpGTID := cp.cfg.EnableGTID && !(oldLocation.GTIDSetStr() == "" && binlog.ComparePosition(oldLocation.Position, binlog.MinPosition) > 0)
```

Comment:
why  `binlog.ComparePosition(oldLocation.Position, binlog.MinPosition) < 0` will happen ?

Reply:
This check is to handle the case that we update enable-gtid = false to true. We only adjust gtid for global checkpoint but omit table checkpoint now. `IsOlderThanTablePoint` will compare two gtidSetStr so that we won't update table point any more.
`binlog.ComparePosition(oldLocation.Position, binlog.MinPosition) > 0` check is used to check whether this table has position-based checkpoint before.